pkgs/top-level/all-packages.nix

```diff
@@ -19699,7 +19699,7 @@ in
     # udev is the same package as systemd which depends on cryptsetup
     # which depends on lvm2 again.  But we only need the libudev part
     # which does not depend on cryptsetup.
-    udev = systemdMinimal;
+    udev = if stdenv.hostPlatform.isMusl then null else systemdMinimal;
```

Comment:
Overwriting packages with null is not great.

Reply:
Have a look at lvm2, it does different things depending if udev is null. Since when is it not acceptable to override it like this?

Reply:
It should be replaced with a boolean flag. If we remove udev entirely we won't necessarily notice that it was used there.